repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -1063,6 +1053,16 @@ private Map<String, Object> preloadProperties(AtlasVertex entityVertex, AtlasEnt
         return propertiesMap;
     }
 
+    private void updateAttrValue( Map<String, Object> propertiesMap, VertexProperty<Object> property){
+        Object value = propertiesMap.get(property.key());
+        if (value instanceof List) {
+            ((List) value).add(property.value());
+        } else {
+            List<Object> values = new ArrayList<>();
+            values.add(property.value());
+            propertiesMap.put(property.key(), values);
```

Comment:
above is not the same as 
` List<Object> values = new ArrayList<>();
                            values.add(propertiesMap.get(property.key()));
                            values.add(property.value());
                            propertiesMap.put(property.key(), values);`